pygments/lexers/macaulay2.py

```diff
@@ -0,0 +1,1661 @@
+    "RunExternalM2",
+    "Saturation",
+    "Schubert2",
+    "SchurComplexes",
+    "SchurFunctors",
+    "SchurRings",
+    "scriptCommandLine",
+    "SCSCP",
+    "SectionRing",
+    "SeeAlso",
+    "SegreClasses",
+    "SemidefiniteProgramming",
+    "Seminormalization",
+    "SeparateExec",
+    "Serialization",
+    "sheafExt",
+    "ShimoyamaYokoyama",
+    "showClassStructure",
+    "showStructure",
+    "showUserStructure",
+    "SimpleDoc",
+    "SimplicialComplexes",
+    "SimplicialDecomposability",
+    "SimplicialPosets",
+    "SimplifyFractions",
+    "SizeLimit",
+    "SkewCommutative",
+    "SlackIdeals",
+    "SLnEquivariantMatrices",
+    "SLPexpressions",
+    "Sort",
+    "SortStrategy",
+    "SourceCode",
+    "SourceRing",
+    "SpaceCurves",
+    "SparseResultants",
+    "SpechtModule",
+    "SpecialFanoFourfolds",
+    "SpectralSequences",
+    "SRdeformations",
+    "Standard",
+    "StartWithOneMinor",
+    "StatePolytope",
+    "StatGraphs",
+    "stderr",
+    "stdio",
+    "StopBeforeComputation",
+    "stopIfError",
+    "StopWithMinimalGenerators",
+    "Strategy",
+    "StronglyStableIdeals",
+    "Style",
+    "SubalgebraBases",
+    "Subnodes",
+    "SubringLimit",
+    "subscript",
+    "Sugarless",
+    "SumsOfSquares",
+    "SuperLinearAlgebra",
+    "superscript",
+    "SVDComplexes",
+    "SwitchingFields",
+    "SymbolicPowers",
+    "SymmetricPolynomials",
+    "Synopsis",
+    "Syzygies",
+    "SyzygyLimit",
+    "SyzygyMatrix",
+    "SyzygyRows",
+    "TangentCone",
+    "TateOnProducts",
+    "TensorComplexes",
+    "Test",
+    "testExample",
+    "TestIdeals",
+    "TeXmacs",
+    "Text",
+    "ThreadedGB",
+    "Threshold",
+    "Topcom",
+    "topLevelMode",
+    "Tor",
+    "TorAlgebra",
+    "Toric",
+    "ToricInvariants",
+    "ToricTopology",
+    "ToricVectorBundles",
+    "TotalPairs",
+    "Tree",
+    "TriangularSets",
+    "Tries",
+    "Trim",
+    "Triplets",
+    "Tropical",
+    "true",
+    "Truncate",
+    "Truncations",
+    "TypicalValue",
+    "typicalValues",
+    "Undo",
+    "Unique",
+    "Units",
+    "Unmixed",
+    "Up",
+    "UpdateOnly",
+    "UpperTriangular",
+    "Usage",
+    "UseCachedExampleOutput",
+    "UseHilbertFunction",
+    "UserMode",
+    "UseSyzygies",
+    "Variable",
+    "VariableBaseName",
+    "Variables",
+    "Vasconcelos",
+    "VectorFields",
+    "VectorGraphics",
+    "Verbose",
+    "Verbosity",
+    "Verify",
+    "VersalDeformations",
+    "Version",
+    "version",
+    "VerticalSpace",
+    "viewHelp",
+    "VirtualResolutions",
+    "Visualize",
+    "WebApp",
+    "Weights",
+    "WeylAlgebra",
+    "WeylGroups",
+    "Wrap",
+    "XML"
+    )
+
+class Macaulay2Lexer(RegexLexer):
+    name = 'Macaulay2'
+    aliases = ['macaulay2']
+    filenames = ['*.m2']
+
+    tokens = {
+        'root': [
+            (r'--.*$', Comment.Single),
+            (r'-\*[\w\W]*?\*-', Comment.Multiline),
+            (r'".*?"', String),
```

Comment:
No idea!   Fixed in https://github.com/pygments/pygments/pull/1791/commits/6f8a0d1411a269e70d1b6fcdfe9269a0ff849eb5

Reply:
Actually, now I remember.  We want lazy matching here.  For example, each line below should be contain two separate comments/strings:

```m2
-* foo *- 2 -* bar *-
"foo" | "bar"
///foo/// | ///bar///
```

Removing my earlier fix.

Reply:
Can you please make sure this is covered in the test case?

Reply:
Also, this is a super error prone pattern I've seen a bunch of times. I'd suggest to implement a proper string state, because as soon as there is string escaping (which is usually inevitable) this becomes a mess. Basically you enter a string case when you see a ", you exit when you see it again, and within the string case you accept anything but ".

Reply:
Ok, I'll try that out!  Thanks for the suggestion.

Reply:
I've added states for strings and block comments, replacing the lazy matching, and have added accompanying tests, too.